### DISCUSSION TITLE
Best practice for borg + rclone

### DISCUSSION THREAD

**Discussion by kaibagley** [nextcloud/all-in-one#2917] [Questions] [answered]
Hey, maybe my pea brain isn't reading the README properly, but is it fine to simply rclone the borg backup folder to google drive or whatever other remote location? I'm seeing that Borg encrypts everything, as you get given a key on the AIO page, but will I be needing anything else? i.e., if I installed AIO on another computer, all I need to do is point it at the Borg backup, give it the key and bam my AIO is back to what it was before?

Thanks!

**Comment by Zoey2936** [comment, marked as answer]
see https://github.com/nextcloud/all-in-one#are-remote-borg-backups-supported

**Reply by dantefromhell** [reply]
I'm starting `rclone` as a service with a read/ write vfs cache. It accelerates the overall backup process for the (IMO minor) downside that the data sync only finishes some time later.
Hence a successful backup notification does not imply a successful transfer - if this is important to you, this solution is not suitable.

My systemd service configuration looks like this:
```
[Unit]
Description=RClone StorJ mount
Wants=network-online.target
After=network-online.target

[Service]
Type=notify
Environment=RCLONE_CONFIG=/etc/rclone/storj.conf
#KillMode=none
RestartSec=5
# rclone is in this strange path due to Ansible role 'stefangweichinger.ansible-rclone'
ExecStart=/usr/local/bin/rclone mount storj:nextcloud/backups /var/cache/rclone/storj-backups \
# StorJ backend does not support polling
--poll-interval 0 \
# Default stats interval is 1min - reduce rate to keep log file smaller
--stats 5m \
# Log to systemd/ journald
--log-systemd --log-level INFO \
# Let only root access the folder
--umask 077 \
# Setup remote-control port on localhost w/o auth
--rc --rc-addr 127.0.0.1:5572 --rc-no-auth \
# VFS caching enables normal filesystem behaviour which is great, but limit cache size
--vfs-cache-mode full --cache-dir /var/cache/rclone/storj-backups \
--vfs-cache-max-size 15G \
# Make tools like 'df' report consumed space correctly
--vfs-used-is-size
ExecStop=/usr/bin/fusermount -u /backups
# Force a VFS cache refresh after daemon start
ExecStartPost=/usr/local/bin/rclone rc vfs/refresh recursive=true _async=true
Restart=on-failure

[Install]
WantedBy=multi-user.target
```

I use `storj` as a storage backend, the file `/etc/rclone/storj.conf` only contains access credentials.

This setup requires some free disc space for the cache, I cap it to 15GB. As borg is writing to a disc backed write cache, the setup accelerates the backup process quite a bit as the limiting factor is the speed of the disc the cache is located on.

If borg is writing more than >15GB though, the write speed will reduce to the upload rate as new writes into the cache only succeed after space has been freed by uploading the data.
